### PR TITLE
search and filter

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine
- https://edu-redmine.sun-asterisk.vn/issues/73187

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image
image
image
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/Order.controller.ts [hunk 1]
```diff
@@ -301,3 +301,81 @@ export const postChangeStatusOrder = async (
...
+    });
+    return;
+  } catch (err) {
+    console.error(err);
+    next();
+  }
+}
+
+// GET filter date
+export const getFilterDate = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (huongnt-2545) [inline src/controllers/Order.controller.ts]: sao chỗ này phải tách thành 2 function khác nhau để xử lý search ở màn list e nhỉ?

File: src/public/css/success.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+body { text-align: center; padding: 40px 0; background: #EBF0F5; } h1 {
+  color: #88B04B; font-family: "Nunito Sans", "Helvetica Neue", sans-serif;
+  font-weight: 900; font-size: 40px; margin-bottom: 10px; } p { color:
+  #404F5E; font-family: "Nunito Sans", "Helvetica Neue", sans-serif;
+  font-size:20px; margin: 0; } i { color: #9ABC66; font-size: 100px;
+  line-height: 200px; margin-left:-15px; } .card { background: white; padding:
+  60px; border-radius: 4px; box-shadow: 0 2px 3px #C8D0D8; display:
+  inline-block; margin: 0 auto; }
```
Comment (huongnt-2545) [inline src/public/css/success.css]: format file css

**Comment by ghost** [comment]
ready

File: src/public/css/success.css (hunk 2 above)
Comment (ghost) [reply src/public/css/success.css]: vâng ạ!

File: src/controllers/Order.controller.ts (hunk 1 above)
Comment (ghost) [reply src/controllers/Order.controller.ts]: Em sẽ gộp lại ạ!

File: src/controllers/Order.controller.ts [hunk 3]
```diff
@@ -301,3 +301,57 @@ export const postChangeStatusOrder = async (
...
+        queryBuilder = queryBuilder.andWhere('order.status = :status', { status });
+      }
+    } else if (req.query.dateInput) {
+      const dateForm = req.query.dateInput;
+      filterCondition = `dateInput=${dateForm}`;
+      queryBuilder = queryBuilder.andWhere('order.createdAt like :date', { date: `%${dateForm}%` });
+    } else {
+      res.redirect('/');
+      return;
+    }
```
Comment (huongnt-2545) [inline src/controllers/Order.controller.ts]: xử lý lại để có thể search + filter đồng thời

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**